[CAS-1229] Fixing state for anonymous user (#2255)
GetStream/stream-chat-android: UNRELEASED_CHANGELOG.md

## stream-chat-android-ui-common
### 🐞 Fixed

> Bug related to anonymous users when app goes onPause and then comes back to onResume state. 

Reviewer: ```suggestion
- Fixed a bug in state handling for anonymous users. 
```

Bullet points! (+ only useful details)
